File: stress-tests/python/mixed.py
"""
Patrick May
Junior IS
Thrashing Workload Script
"""
from wrapper import measure_time

def thrash():
    counter = 0
    while counter < 10_000_000:
        counter += 1
    counter = 0

@measure_time
def turbo_thrash():
    import threading

    for c in range(20):
        threading.Thread(thrash())

@measure_time
def think():
    from time import sleep
    sleep(10)

for _ in range(6):
    turbo_thrash() if _ % 2 else think()

File: stress-tests/c++/mixed.cpp
#include <bits/stdc++.h>
#include <unistd.h>

#include "wrapper.hpp"

using namespace std;


void snooze(){
    CowLog s("snooze");
        
    sleep(10);

    s.finish();
}

void thrash(){
    CowLog s("thrash");

    long long int ct = 0;
    for(long long int i = 0; i < 1000000000; ++i){
        ct += 1;
    }

    s.finish();
}
int main() {
    cout << "thonk time 🤬" << "\n";
    for(int i = 0; i < 10; ++i){
        if(i % 2){
            thrash();
        }
        else{
            snooze();
        }
    }
}